File: Data Structures & Algorithms/validate-parentheses/submission-1.java
class Solution {
    public boolean isValid(String s) {
      Stack<Character> stack = new Stack<>();
      for(int i = 0; i < s.length(); i++){
        char var = s.charAt(i);
        
        if(var == '(' || var == '[' || var == '{'){
            stack.push(var);
        }

        else if(var == ')' || var == ']' || var == '}'){
            if(stack.isEmpty())  return false; 
            char val = stack.pop();

            if((var == ')' && val != '(') ||
               (var == ']' && val != '[') ||
               (var == '}' && val != '{')) {
                return false;
              }
        }
      }
    return stack.isEmpty();
    }
}


        

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        if not root:
            return None
        
        heap = []
        stack = [root]

        while stack:
            node = stack.pop()

            heapq.heappush(heap, -node.val)

            if len(heap) > k: heapq.heappop(heap)

            if node.right:
                stack.append(node.right)
            if node.left:
                stack.append(node.left)
        
        return -(heapq.heappop(heap))

            